### PR TITLE
add BFloat16 support for sparse.addmm on CPU

### PR DESCRIPTION
**Pull request by jiayisunx**
Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by samdow** [comment]
@cpuhrsch adding for sparsity, feel free to tag someone else if that's not right

**Comment by cpuhrsch** [comment]
@samdow - I'm the right POC, thanks for tagging.

**Review by cpuhrsch — CHANGES_REQUESTED** [review]
@jiayisunx - please also extend test coverage for this type :)

**Comment by jiayisunx** [comment]
> @jiayisunx - please also extend test coverage for this type :)

have added UT, please help review it again.

File: test/test_sparse.py [hunk 1]
```diff
@@ -1283,6 +1283,25 @@ def fn(S, D1, D2, beta=beta, alpha=alpha):
         test_shape(7, 8, 9, 20, False, (1, 1))
         test_shape(7, 8, 9, 20, True, (1, 1))
 
+    @onlyCPU
+    @coalescedonoff
+    def test_sparse_addmm_bfloat16(self, device, coalesced):
```
Comment (cpuhrsch) [inline test/test_sparse.py:1288]: What stands in the way of extending above test test_sparse_addmm by adding bfloat16 to the dtypes decorator?

**Review by cpuhrsch — CHANGES_REQUESTED** [review]
I'm worried the one-off test isn't necessary

File: test/test_sparse.py (hunk 1 above)
Comment (jiayisunx) [reply test/test_sparse.py:1288]: I have tried to add UT by this way, but the test brings in five more unsupported OPs.

Comment (cpuhrsch) [reply test/test_sparse.py:1288]: Would be equally as easy to add bfloat16 support for those as well as in this case?

Comment (cpuhrsch) [reply test/test_sparse.py:1288]: Yes, but you could change that precision for bfloat16 specifically. You can branch on the dtype input and set a different precision for comparison.

**Pull request closed without merging**